hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -659,22 +667,34 @@ async def _listfiles_flat(self, bucket: str, name: str) -> AsyncIterator[FileLis
                 for prefix in prefixes:
                     assert prefix.endswith('/')
                     url = f'gs://{bucket}/{prefix}'
+                    if not force_all and await self.isfile(url):
```

Comment:
I think this is wrong.  We should return directories that contain only empty files.

Reply:
See my comment above. For `rmtree` where `force_all` is True, then we do want to return non-empty directory/file files.